### PR TITLE
Fix shape mismatch in Keras Attention layer during masking

### PR DESCRIPTION
**Pull request by sonali-kumari1**
This PR fixes the `RuntimeError` encountered in the Attention layer when using masks. The error arises from a shape mismatch between the mask and attention scores, resulting in a broadcasting issue during training.
Fixes: [#21483 ](https://github.com/keras-team/keras/issues/21483)

### CODE REVIEW

**Comment by sonali-kumari1** [comment]
Hi @fchollet @hertschuh - 
I have added a unit test  to verify shape mismatch with 2D mask. Thanks!

File: keras/src/layers/attention/attention_test.py [hunk 1]
```diff
@@ -86,6 +86,21 @@ def test_attention_with_mask(self):
         self.assertAllClose(output, [[[1.0, 1.0], [0.0, 0.0]]])
         self.assertAllClose(scores, [[[1.0, 0.0], [1.0, 0.0]]])
 
+    def test_attention_2D_mask_shape_mismatch(self):
+        layer = layers.Attention()
+        batch_size, Tq, Tv, dim = 2, 3, 3, 4
```
Comment (hertschuh) [inline keras/src/layers/attention/attention_test.py]: Please use different values for `Tq` and `Tv` so that the tests check that there is no mixup.

File: keras/src/layers/attention/attention_test.py [hunk 2]
```diff
@@ -88,11 +88,13 @@ def test_attention_with_mask(self):
 
     def test_attention_2D_mask_shape_mismatch(self):
         layer = layers.Attention()
-        batch_size, Tq, Tv, dim = 2, 3, 3, 4
+        batch_size, Tq, Tv, dim = 2, 3, 4, 4
```
Comment (hertschuh) [inline keras/src/layers/attention/attention_test.py]: Sorry to be bother you again about this, but can you make all values unique? Like use `5` for `dim`.

Again, we want to make sure dimensions are matched up correctly in unit tests.

Comment (sonali-kumari1) [reply keras/src/layers/attention/attention_test.py]: @hertschuh, I updated the unit test to use distinct values for `batch_size`, `Tq`, `Tv` and `dim` to properly detect shape mismatches. Thank you!

**Review by hertschuh — APPROVED** [review]
Thank you for the fix!
